From 7b031dd95fab3afdaeb8dd3f10ebe9b206b4dfd2 Mon Sep 17 00:00:00 2001
From: ratchaya <quic_ratchaya@quicinc.com>
Date: Wed, 23 Aug 2023 23:16:04 +0530
Subject: [PATCH] firewall4: Add firewall disabled option in config

Add disabled option to help while transfering file
using TFTP, avoiding firewall restart by network restart.

Change-Id: I6207f16309221e5b30b4c6895e73bfaff610a64c
Signed-off-by: ratchaya <quic_ratchaya@quicinc.com>
---

Index: firewall4-2023-03-23-04a06bd7/root/etc/config/firewall
===================================================================
--- firewall4-2023-03-23-04a06bd7.orig/root/etc/config/firewall
+++ firewall4-2023-03-23-04a06bd7/root/etc/config/firewall
@@ -5,6 +5,7 @@ config defaults
 	option forward		REJECT
 # Uncomment this line to disable ipv6 rules
 #	option disable_ipv6	1
+	option disabled         0
 
 config zone
 	option name		lan
Index: firewall4-2023-03-23-04a06bd7/root/sbin/fw4
===================================================================
--- firewall4-2023-03-23-04a06bd7.orig/root/sbin/fw4
+++ firewall4-2023-03-23-04a06bd7/root/sbin/fw4
@@ -17,6 +17,8 @@ die() {
 }
 
 start() {
+	disabled=$(uci get firewall.@defaults[0].disabled)
+	[ "$disabled" -eq 1 ] && return 0
 	{
 		flock -x 1000
 
